apps/server/src/modules/class/repo/classes.repo.ts

```diff
@@ -86,4 +57,13 @@ export class ClassesRepo {
 
 		return domainObject;
 	}
+
+	public async removeUserReference(userId: EntityId): Promise<number> {
+		const id = new ObjectId(userId);
+		const count = await this.em.nativeUpdate(ClassEntity, { $or: [{ userIds: id }, { teacherIds: id }] }, {
```

Comment:
Not big part. But we have a mix of orchestration and logic. 
Building the query is the logic part. We have the not well named Scope classes, maybe better it should named Class(MongoDb)QueryBuilder. That can do the logic part.

Reply:
I would keep this as an excpetion. Only in the case of this nativeUpdate, is best to see the query well. These queries with Partial and field names are difficult to handle.
To move only this method in a separate class and inject that also in the service, is perhaps too much code splitting at this stage.